### ISSUE TITLE
Content which should be hidden is exposed to accessibility

### ISSUE THREAD

**Issue by jcsteh** [mozilla-iam/auth0-custom-lock#76]
STR (with Firefox and the NVDA screen reader):

1. Visit a Mozilla site which requires you to log in. You will be focused in the E-mail address field.
2. Switch to browse mode with NVDA+space (insert+space by default).
3. Repeatedly press down arrow to read the content on the page after the email address field.
    - Expected: You should hear the "Enter" button associated with the email address field, but not much other content.
    - Actual: You hear content from other screens which are visually hidden; e.g. a password field, success and error messages from other login methods, etc.

Ideally, this hidden content would be CSS display: none, but I'm guessing there's a reason something else is being used. In that case, the aria-hidden="true" attribute will need to be used to hide the hidden content for accessibility purposes (and obviously the attribute needs to be removed for content as it is dynamically shown).

**Comment by MarcoZehe** [comment]
This seems to hae crept in late, since I don't recall this to be an issue when @hidde and I tested this on the staging site. :( Yeah, display:none; or visibility:hidden; are definitely the preferred ways here, aria-hidden should be the last measure. Just moving things out of the view port doesn't hide them from screen readers.

**Comment by hidde** [comment]
Thanks for reporting. 

Basically, the story is: we did have `display: none` in place on all the hidden ‘cards’, as we should have in this case. 

However, this made password managers, notably LastPass, disregard the login form, so that most of its features didn’t work. Because LastPass is our recommended password manager, I worked to find a way that still properly makes cards unavailable to AT, yet available to LastPass. 

I believe even `aria-hidden` made the password field unavailable to LastPass, but I will investigate. 

Again, thanks for reporting!

**Comment by hidde** [comment]
I have just pushed a fix for this, we now use `aria-hidden` in addition to the existing ‘hacky’ hiding, so that hidden screens are hidden from everyone including AT users, but except the LastPass algorithm (sigh 😒)

**Comment by jcsteh** [comment]
Unfortunately, this doesn't work with NVDA. I noticed this some time ago, but didn't get around to investigating it until today. :(

It looks like the code sets aria-hidden, but doesn't explicitly set `aria-hidden="true"`; i.e. it does `<div aria-hidden ...>`, but must instead do `<div aria-hidden="true" ...>`. Unlike normal boolean HTML attributes such as `checked` or `hidden`, [boolean ARIA attributes in HTML require a "true" or "false" string](https://www.w3.org/WAI/PF/aria-1.1/appendices#typemapping). I guess VoiceOver doesn't require this for some reason, but that usage isn't spec compliant and doesn't work in Firefox.

**Comment by hidde** [comment]
Woa, today I learned! Thanks for spotting this, I will update it.

**Comment by hidde** [comment]
This has been fixed, it will be updated in the next release of the login form.
